components/region_cache_memory_engine/src/background.rs

```diff
@@ -37,11 +43,100 @@ fn parse_write(value: &[u8]) -> Result<WriteRef<'_>, String> {
     }
 }
 
+// BgWorkManager managers the worker inits, stops, and task schedules. When
+// created, it starts a worker which receives tasks such as gc task, range
+// delete task, range snapshot load and so on, and starts a thread for
+// periodically schedule gc tasks.
+pub struct BgWorkManager {
+    worker: Worker,
+    scheduler: Scheduler<BackgroundTask>,
+    tick_stopper: Option<(JoinHandle<()>, Sender<bool>)>,
+}
+
+impl Drop for BgWorkManager {
+    fn drop(&mut self) {
+        let (h, tx) = self.tick_stopper.take().unwrap();
+        let _ = tx.send(true);
+        let _ = h.join();
+        self.worker.stop();
+    }
+}
+
+impl BgWorkManager {
+    pub fn new(core: Arc<ShardedLock<RangeCacheMemoryEngineCore>>, gc_interval: Duration) -> Self {
+        let worker = Worker::new("range-cache-background-worker");
+        let runner = BackgroundRunner::new(core.clone());
+        let scheduler = worker.start("range-cache-engine-background", runner);
```

Comment:
why not let this gc runner running in the tikv background worker

Reply:
Let it transparent for raftstore.